### PR TITLE
Upgrade to lucene-6.4.0-snapshot-084f7a0.

### PR DESCRIPTION
**Pull request by jpountz**
This new snapshot has the synonym graph filters and new helpers for
exclusive bounds on ranges. @mattweber @mikemccand Could you
have a look at the synonym part?

### CODE REVIEW

**Comment by s1monw** [comment]
LGTM

**Comment by jpountz** [comment]
@elasticmachine retest this please

File: core/src/main/java/org/elasticsearch/index/search/MatchQuery.java [hunk 1]
```diff
@@ -386,7 +385,7 @@ protected final Query createFieldQuery(TokenStream source, BooleanClause.Occur o
                     }
 
                     int positionLength = posLenAtt.getPositionLength();
-                    if (!isGraph && positionLength > 1 && ((flagsAtt.getFlags() & GRAPH_FLAG) == GRAPH_FLAG)) {
+                    if (positionLength > 1) {
```
Comment (mattweber) [inline core/src/main/java/org/elasticsearch/index/search/MatchQuery.java]: Don't need `TypeAttribute` and if LUCENE-7603 is here we can remove all the Overrides on `QueryBuilder` that were added for graph support.

Comment (mikemccand) [inline core/src/main/java/org/elasticsearch/index/search/MatchQuery.java]: > Don't need TypeAttribute and if LUCENE-7603 is here we can remove all the Overrides on QueryBuilder that were added for graph support.

+1

**Comment by mikemccand** [comment]
LGTM

File: core/src/main/java/org/elasticsearch/index/search/MatchQuery.java (hunk 1 above)
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/index/search/MatchQuery.java]: The snapshot I have does not have the new query builder yet, I'll build a new one.

**Comment by jpountz** [comment]
@mattweber Can you look again please?

**Comment by mattweber** [comment]
@jpountz LGTM

**Pull request merged**